### PR TITLE
[BPF/probes]: Added kernel checks Fixes: #891

### PR DESCRIPTION
**Pull request by eloycoto**
Hello!

This PR check the kernel config and validate that the minimum kernel params are satisfied. 

CoreOS uses /proc/config, but the other distros are using "/boot-config-${kernel-release}". I tested with major distributions and should work ok in all of them. 

On the other hand, maybe some modules should be checked too, like "algif_hash" but I was not entirely sure, and I didn't add it. If you want to add, I'll do. 

Thanks!

### CODE REVIEW

File: bpf/run_probes.sh [hunk 1]
```diff
@@ -27,51 +27,92 @@ FEATURE_FILE="$RUNDIR/globals/bpf_features.h"
...
+        zgrep "${key}" $KCONFIG > /dev/null || {
+            RESULT=1;
+            echo "WARNING: BPF/probes: ${key} is not in kernel configuration" >> $WARNING_FILE
+            }
+    done
+
+    if [[ "$RESULT" -gt 0 ]]; then
+        echo "Error: BPF/probes: No valid kernerl configuration" >> $WARNING_FILE
+        return
+    fi
```
Comment (aalemayhu) [inline bpf/run_probes.sh]: Is this return needed?

File: bpf/run_probes.sh [hunk 2]
```diff
@@ -27,51 +27,92 @@ FEATURE_FILE="$RUNDIR/globals/bpf_features.h"
...
-	if [ ! -z "$PROBE_DIR" ]; then
-		rm -rf "$PROBE_DIR"
-	fi
+    if [ ! -z "$PROBE_DIR" ]; then
+        rm -rf "$PROBE_DIR"
+    fi
 
-	#ip link del $DEV 2> /dev/null
+    #ip link del $DEV 2> /dev/null
 }
```
Comment (aalemayhu) [inline bpf/run_probes.sh]: If there is no reason to change the indentation, please leave as is.

**Review by aalemayhu — CHANGES_REQUESTED** [review]
A part from nitpicks, looks good.

File: bpf/run_probes.sh [hunk 3]
```diff
@@ -27,51 +27,92 @@ FEATURE_FILE="$RUNDIR/globals/bpf_features.h"
...
+    local RESULT=0
+    # Coreos kernel config is on /proc/config (module configs).
+    # Other distros in /boot/config-*
+    local config_locations=("/proc/config" "/proc/config.gz",
+        "/boot/config-$(uname -r)")
+    local PARAMS=(
+        "CONFIG_CGROUP_BPF=y" "CONFIG_BPF=y" "CONFIG_BPF_SYSCALL=y"
+        "CONFIG_NET_SCH_INGRESS=m" "CONFIG_NET_CLS_BPF=m" "CONFIG_NET_CLS_ACT=y"
+        "CONFIG_BPF_JIT=y" "CONFIG_LWTUNNEL_BPF=y" "CONFIG_HAVE_EBPF_JIT=y"
+        "CONFIG_BPF_EVENTS=y" "CONFIG_TEST_BPF=m")
```
Comment (tgraf) [inline bpf/run_probes.sh]: Several of these can be configured as both a module or builtin. I think these must be regular expressions which we pass into `zgrep -E` or something similar.

**Review by tgraf — CHANGES_REQUESTED** [review]

File: bpf/run_probes.sh [hunk 4]
```diff
@@ -36,6 +36,47 @@ function cleanup {
...
+    for key in "${PARAMS[@]}"
+    do
+        zgrep -E "${key}" $KCONFIG > /dev/null || {
+            RESULT=1;
+            echo "WARNING: BPF/probes: ${key} is not in kernel configuration" >> $WARNING_FILE
+            }
+    done
+
+    if [[ "$RESULT" -gt 0 ]]; then
+        echo "Error: BPF/probes: No valid kernerl configuration" >> $WARNING_FILE
```
Comment (tklauser) [inline bpf/run_probes.sh]: Typo: should be _kernel_ instead of _kernerl_

**Review by aalemayhu — APPROVED** [review]
Requires re-base, but otherwise looks good.

**Review by tgraf — APPROVED** [review]

**Pull request merged**